### PR TITLE
koordlet: enhance node prediction

### PR DESCRIPTION
**Pull request by saintube**
### Ⅰ. Describe what this PR does

koordlet: enhance the node prediction.

- Add the node priority predictor and the min predictor. s.t.

```
Reclaimable[Prod] := max(0, min(Requests[Prod] - Peak[Prod], sum(Request[ProdPod] - Peak[ProdPod])))
```

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

File: pkg/util/resource.go [hunk 1]
```diff
@@ -44,6 +44,26 @@ func MultiplyQuant(quant resource.Quantity, factor float64) resource.Quantity {
 	return *newQuant
 }
 
+// Min returns the result of Min(a, b) for each named resource, corresponded to the quotav1.Max().
+func Min(a corev1.ResourceList, b corev1.ResourceList) corev1.ResourceList {
```
Comment (hormes) [inline pkg/util/resource.go:48]: The semantics of this function are a bit strange, such as A is {cpu = 1}, B is {memory = 1}, after Min is {cpu = 1, memory = 1}. For quotav1.Max, union keys is semantic, but min is a bit problematic.

File: pkg/koordlet/prediction/peak_predictor.go [hunk 2]
```diff
@@ -65,18 +68,33 @@ func NewPredictorFactory(predictServer PredictServer, coldStartDuration time.Dur
 func (f *predictorFactory) New(t PredictorType) Predictor {
 	switch t {
 	case ProdReclaimablePredictor:
-		return &prodReclaimablePredictor{
+		podPredictor := &podReclaimablePredictor{
```
Comment (hormes) [inline pkg/koordlet/prediction/peak_predictor.go:71]: It is elegant to implement the combination of two predictors here.

File: pkg/koordlet/prediction/peak_predictor.go [hunk 3]
```diff
@@ -65,18 +68,33 @@ func NewPredictorFactory(predictServer PredictServer, coldStartDuration time.Dur
...
 			reclaimable:         util.NewZeroResourceList(),
 			pods:                make(map[string]bool),
 		}
+		priorityPredictor := &priorityReclaimablePredictor{
+			predictServer:         f.predictServer,
+			safetyMarginPercent:   f.safetyMarginPercent,
+			priorityClassFilterFn: isPriorityClassReclaimableForProd,
+			reclaimRequest:        util.NewZeroResourceList(),
+		}
+		return &minPredictor{
```
Comment (hormes) [inline pkg/koordlet/prediction/peak_predictor.go:85]: It is elegant to implement the combination of two predictors here.

File: pkg/util/resource.go (hunk 1 above)
Comment (saintube) [reply pkg/util/resource.go]: Fixed. Now the unmatched resource names will be dropped in the result.
e.g. a = {"cpu": "1"}, b = {"memory", "2Gi"} => {}

**Pull request merged**